api/agent/drivers/docker/imageCache.go

```diff
@@ -0,0 +1,108 @@
+package docker
+
+import (
+	"container/list"
+	"sync"
+
+	d "github.com/fsouza/go-dockerclient"
+)
+
+type ImageEvictor func(d.APIImages) error
+
+// Cache is an LRU cache, safe for concurrent access.
+type Cache struct {
+	totalSize int64
+	mu        sync.Mutex
+	ll        *list.List
+	cache     map[string]*list.Element
+	maxSize   int64
+	onEvict   ImageEvictor
+}
+
+// *entry is the type stored in each *list.Element.
+type entry struct {
+	key   string
+	value d.APIImages
+}
+
+// New returns a new cache with the provided maximum items.
+func NewLRU(maxSize int64, onEvict ImageEvictor) *Cache {
+	return &Cache{
+		ll:      list.New(),
+		cache:   make(map[string]*list.Element),
+		onEvict: onEvict,
```

Comment:
maxSize not initialized? I think we also (here or at a higher layer in driver code) need to handle the case where maxSize is zero. That should be unlimited where evictions get disabled.